CRM/Contribute/BAO/Query.php

```diff
@@ -1197,4 +1198,30 @@ public static function buildDateWhere(&$values, $query, $name, $field, $title) {
     return TRUE;
   }
 
+  /**
+   * Check if the values in the date range are in correct chronological order.
+   *
+   * @param array $fields
+   * @param array $files
+   * @param CRM_Core_Form $form
+   *
+   * @return bool|array
+   */
+  public static function formRule($fields, $files, $form) {
+    $errors = array();
+
+    if (empty($fields['contribution_date_high']) || empty($fields['contribution_date_low'])) {
+      return TRUE;
+    }
+    $lowDate = strtotime($fields['contribution_date_low']);
+    $highDate = strtotime($fields['contribution_date_high']);
+
+    if ($lowDate > $highDate) {
+      $errors['contribution_date'] = ts('Please check your Contribution Date Range.');
+      // remove this after errors starts working.
+      echo ts('Please check your Contribution Date Range.');
+    }
+    return empty($errors) ? TRUE : $errors;
```

Comment:
The function works properly (checked with the test echo statement) however the errors do not show up.